providers/decentholograms/src/main/java/de/minebench/tresor/providers/decentholograms/DecentHologramsProvider.java

```diff
@@ -0,0 +1,48 @@
+package de.minebench.tresor.providers.decentholograms;
+
+import de.minebench.tresor.Provider;
+import de.minebench.tresor.services.hologram.Hologram;
+import de.minebench.tresor.services.hologram.HologramProvider;
+import eu.decentsoftware.holograms.api.DHAPI;
+import eu.decentsoftware.holograms.plugin.DecentHologramsPlugin;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+
+public class DecentHologramsProvider extends Provider<HologramProvider, DecentHologramsPlugin> implements HologramProvider {
```

Comment:
Thanks for providing an example implementation which wasn't really part of the issue! However this needs to be formatted slightly differently: Providers are [automatically loaded by ProviderManagers](https://github.com/Minebench/Tresor/blob/b85cc67ef1637561b4cb29fe1f1dec8bc15912da/plugin/src/main/java/de/minebench/tresor/Tresor.java#L72-L78) using my [Hook library](https://github.com/Phoenix616/hook). Part of the design is that certain class names are used, in this case `<PluginName><ServiceName>` so (assuming the service interface is renamed `Holograms`) this service provider implementation would need to be named `DecentHologramsHolograms` (the doubling is a bit unfortunate but technically nobody is going to see this name anyways... I could've gone with `DecentHologramsHologramsProvider` but that's even more of a mouthfull...).

This is done to support automatic multiple different services for the same plugin ([see Craftconomy3](https://github.com/Minebench/Tresor/tree/b85cc67ef1637561b4cb29fe1f1dec8bc15912da/providers/craftconomy3/src/main/java/de/minebench/tresor/providers/craftconomy3)) without having to split them up into packages in already split modules.

I realise that this design is not documented right now and I am fine adjusting this accordingly myself seeing as actually implementing providers wasn't really meant to be part of the issue itself anyways :)